packages/algob/src/builtin-tasks/sign-multisig.ts

```diff
@@ -0,0 +1,97 @@
+import algosdk from "algosdk";
+import fs from "fs-extra";
+import path from "path";
+
+import { task } from "../internal/core/config/config-env";
+import { ASSETS_DIR } from "../internal/core/project-structure";
+import { RuntimeEnv } from "../types";
+import { TASK_SIGN_MULTISIG } from "./task-names";
+
+export interface TaskArgs {
+  file: string
+  account: string
+  out: string
+}
+
+async function multiSignTx (
+  taskArgs: TaskArgs,
+  runtimeEnv: RuntimeEnv
+): Promise<void> {
+  const accounts = runtimeEnv.network.config.accounts;
+  let flag = false;
+  const signer = {
+    addr: "",
+    sk: new Uint8Array(0)
+  };
+  let signedTxn;
+  for (var account of accounts) {
+    if (account.name === taskArgs.account) {
+      flag = true;
+      signer.addr = account.addr;
+      signer.sk = account.sk;
+    }
+  }
+  if (!flag) {
+    console.error("No account with the name \"%s\" exists in the config file.", taskArgs.account);
+    return;
+  }
+  const p = path.join(ASSETS_DIR, taskArgs.file);
+  const txFile = fs.readFileSync(p);
+  if (txFile === undefined) {
+    throw new Error(`File ${taskArgs.file} does not exist`);
+  }
+  console.log(txFile);
+  const tx = algosdk.decodeObj(txFile);
+  if (!tx.blob) {
+    console.error("The decoded transaction doesn't appear to be a signed transaction.");
+    return;
+  }
+  const decodedTx = algosdk.decodeSignedTransaction(tx.blob);
+  console.debug("Decoded txn from %s: %O", taskArgs.file, decodedTx);
+  console.log(decodedTx.msig);
+  if (decodedTx.msig) {
+    const addresses = [];
+    for (var sig of decodedTx.msig.subsig) {
+      addresses.push(algosdk.encodeAddress(Uint8Array.from(sig.pk)));
+    }
+    console.log(addresses);
+    const mparams = {
+      version: decodedTx.msig.v,
+      threshold: decodedTx.msig.thr,
+      addrs: addresses
+    };
+    signedTxn = algosdk.appendSignMultisigTransaction(tx.blob, mparams, signer.sk);
```

Comment:
Not sure if that's the right way. I just want to append a signature to a `multisig`. So, i don't think we need `mparams` here. There should be another function in `js-sdk`.

Reply:
I could only find 2 functions. One was to sign a raw txn and one was to append a multisig. In both the cases we need mparams.

Reply:
Alright, i will check as well. Because with `logic signature` we don't require `mparams` when appending to multisig.

Reply:
checked. We can use `appendSignMultisigTransaction` which require mparams